### PR TITLE
Re-jig the Timelord status lights bar

### PR DESCRIPTION
**Pull request by mstratford**

### CODE REVIEW

File: src/scripts/timelord.js [hunk 1]
```diff
@@ -356,6 +358,16 @@ window.Timelord = {
 
 		}
 
+		//also set the OB light to success since we get data.studio here and not above.
+
+		if (data['s4power']) {
```
Comment (LordAro) [inline src/scripts/timelord.js]: why is only obs1 covered here? should be return data for obs2 as well?

File: src/scripts/timelord.js [hunk 2]
```diff
@@ -316,7 +316,9 @@ window.Timelord = {
 		for (var i in Timelord.routeobinfo) {
 
 			if (data[i]) {
-				Timelord.setAlert('ob' + i, 'good');
+				if (!document.getElementById("ob" + i).classList.contains('good')) {
```
Comment (LordAro) [inline src/scripts/timelord.js]: This seems like a backwards way to work out whether OB is on air, is there nothing in `data`?

File: src/scripts/timelord.js (hunk 1 above)
Comment (mstratford) [inline src/scripts/timelord.js]: There's two different api endpoints in use, the selector status (only s1,s2 and s4 (OB1) power plus current studio selected) is the one used here. OB 2 only comes through remotestreams api (in your previous comment)

Comment (mstratford) [inline src/scripts/timelord.js]: However remotestreams doesn't know which is on air.

Comment (LordAro) [inline src/scripts/timelord.js]: ok, so s5 is never a thing from the API, but i think this if statement can still be rolled into the above for loop

File: src/scripts/timelord.js [hunk 3]
```diff
@@ -312,27 +315,23 @@ window.Timelord = {
...
-		for (var i in Timelord.routeobinfo) {
-
-			if (data[i]) {
-				Timelord.setAlert('ob' + i, 'good');
-				if (Timelord.routeobinfo[i] !== true &&
-					Timelord.routeobinfo[i] !== false) {
-					clearTimeout(Timelord.routeobinfo[i]);
+		//this function is now only used for OB2, since it's not selectable
+		var i = 5, studioID = "s" + i;
+			if (data["s" + (i-3)]) {
```
Comment (LordAro) [inline src/scripts/timelord.js]: excessive indent?

File: src/scripts/timelord.js [hunk 4]
```diff
@@ -342,18 +341,28 @@ window.Timelord = {
...
 
-		for (var i = 1; i <= 2; i++) {
-
-			if (data['s' + i + 'power']) {
-				if (data.studio == i) {
-					Timelord.setAlert('s' + i, 'good');
-				} else {
-					Timelord.setAlert('s' + i, 'standby');
+		for (var i = 1; i <= 4; i++) {
+			if (i != 3) { //jukebox has no s3power from API
```
Comment (LordAro) [inline src/scripts/timelord.js]: return/break early to reduce indent? especially now there's no else case `if (i == 3) continue;`

**Comment by mstratford** [comment]
Please note: UniversityRadioYork/MyRadio#633

File: src/scripts/timelord.js [hunk 5]
```diff
@@ -307,53 +289,32 @@ window.Timelord = {
...
 	 *
 	 * @param {Object} data
 	 */
 	setStudioPowerLevel: function (data) {
 
-		for (var i = 1; i <= 2; i++) {
-
+		for (var i = 1; i <= 4; i++) {
 			if (data['s' + i + 'power']) {
 				if (data.studio == i) {
```
Comment (LordAro) [inline src/scripts/timelord.js]: ternary for this block?

File: src/scripts/timelord.js [hunk 6]
```diff
@@ -307,53 +289,32 @@ window.Timelord = {
...
 					Timelord.setAlert('s' + i, 'standby');
 				}
-			} else {
+				if (Timelord.studioinfo['s' + i] !== true &&
+					Timelord.studioinfo['s' + i] !== false) {
+					clearTimeout(Timelord.studioinfo['s' + i]);
+				}
+				Timelord.studioinfo[i] = true;
+			// set red disconnection light
+			} else if (Timelord.studioinfo[i] === true) {
```
Comment (LordAro) [inline src/scripts/timelord.js]: i know javascript is javascript, but is all the explicit `=== true` and `=== false` necessary? these values can't be anything other than boolean, right?

File: src/scripts/timelord.js [hunk 7]
```diff
@@ -3,9 +3,11 @@ window.Timelord = {
...
 	_config: null,
 
-	routeobinfo: {
-		s1: false,
-		s2: false
+	studioinfo: {
+		s1: false, //studio1
+		s2: false, //studio2
+		s3: true, //jukebox
+		s4: false //OB
```
Comment (LordAro) [inline src/scripts/timelord.js]: comment saying what the true/false values represent?

Comment (mstratford) [inline src/scripts/timelord.js]: done :)

File: src/scripts/timelord.js (hunk 6 above)
Comment (mstratford) [inline src/scripts/timelord.js]: Just tested this, yeah, doesn't make a difference.

File: src/scripts/timelord.js [hunk 8]
```diff
@@ -301,11 +301,7 @@ window.Timelord = {
 
 		for (var i = 1; i <= 4; i++) {
 			if (data['s' + i + 'power']) {
-				if (data.studio == i) {
-					Timelord.setAlert('s' + i, 'good');
-				} else {
-					Timelord.setAlert('s' + i, 'standby');
-				}
+				(data.studio == i) ? Timelord.setAlert('s' + i, 'good') : Timelord.setAlert('s' + i, 'standby');
```
Comment (LordAro) [inline src/scripts/timelord.js]: put the ternary inside the Timelord.setAlert call. no need to duplicate most of it

**Pull request merged**